python/test/extrinsics/test_universal_microdata_scheme.py

```diff
@@ -0,0 +1,197 @@
+            faithfulness_threshold=0.05,
+            faithfulness_epsilon=1.,
+            with_selection=True,
+            selection_stop_probability=0,
+            selection_epsilon=0,
+            selection_steps=None
+            )
+
+        assert ums_meas.map(1) == 8
+
+        score, output = ums_meas(real_df)
+
+    assert score
+    assert len(output) == 4
+
+
+def _make_max_marginal_abs_error(
+        input_domain: Domain,
+        input_metric: Metric,
+        *,
+        reference_dataset) -> Transformation:
+    """Construct a Transformation that returns the maximal absoulte error over all k-way marginals."""
+
+    np = pytest.importorskip("numpy")
+    pl = pytest.importorskip("polars")
+
+    dp.assert_features("contrib", "floating-point")
+
+    if "LazyFrameDomain" not in str(input_domain.type):
+        raise ValueError("input_domain must be a LazyFrame domain")
+
+    # TODO: assert margin with public_info "keys"
+    # TODO: extract cols (by) from the magin
+    cols = reference_dataset.columns
+
+    # assert input_domain.member(reference_dataset.lazy())
+
+    if input_metric != dp.symmetric_distance():
+        raise ValueError("input metric must be symmetric distance")
+
+    def function(dataset):
+
+        abs_diffs = []
+        for k_way in range(1, len(cols) + 1):
+            for subset in it.combinations(cols, k_way):
+
+                merged_df = (dataset.group_by(subset).len().collect().join(
+                    reference_dataset.group_by(subset).len(),
+                    how="outer",
+                    on=subset,)
+                    .select([
+                        pl.col("len").alias("len_dataset").cast(pl.Int32),
+                        pl.col("len_right").alias("len_reference_dataset").cast(pl.Int32)
+                    ])
+                    .fill_null(0)
+                )
+
+                abs_diffs.append((merged_df["len_dataset"] - merged_df["len_reference_dataset"])
+                                 .abs()
+                                 .to_numpy())
+
+        return np.concatenate(abs_diffs).max()
+
+    return dp.t.make_user_transformation(
+        input_domain,
+        input_metric,
+        dp.atom_domain(T=int),
+        dp.absolute_distance(T=int),
+        function,
+        lambda d_in: d_in,
+    )
+
+
+def _make_max_univariate_rel_error(
+        input_domain: Domain,
+        input_metric: Metric,
+        *,
+        reference_dataset,
+        clipping_upper_bound: float) -> Transformation:
+    """Construct a Transformation that returns the maximal relative error over all 1-way marginals."""
+
+    np = pytest.importorskip("numpy")
+    pl = pytest.importorskip("polars")
+
+    dp.assert_features("contrib", "floating-point")
+
+    if "LazyFrameDomain" not in str(input_domain.type):
+        raise ValueError("input_domain must be a LazyFrame domain")
+
+    # TODO: assert margin with public_info "keys"
+    # TODO: extract cols (by) from the magin
```

Comment:
If these wouldn't block merge, but they'd be nice to have some day, can you file corresponding issues and add the URLs here?